### ISSUE TITLE
Reload plugins without restarting app

### ISSUE THREAD

**Issue by appetizermonster** [appetizermonster/hain#12]

**Comment by notatestuser** [comment]
It's in now - cool. Could we run this automatically on plugin install/update?

**Comment by appetizermonster** [comment]
@notatestuser Yes technically it's possible, but I don't know to unload nodejs modules without closing process, so all the plugins should be reloaded at once and it takes time. 
Reloading plugins after every installing would be boring.

**Comment by TheElf** [comment]
you could instead modify the plugin api to require support for a shutdown or reload method 

this would be where we require plugins to clean up after themselves
